libraries/USBSerial/src/USBSerial.h

```diff
@@ -0,0 +1,83 @@
+/******************************************************************************
+ * The MIT License
+ *
+ * Copyright (c) 2010 Perry Hung.
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *****************************************************************************/
+
+/**
+ * @brief Wirish virtual serial port
+ */
+
+ /*
+  * Arduino srl - www.arduino.org
+  * 2016 Jun 9: Edited Francesco Alessi (alfran) - francesco@arduino.org
+  */
+
+
+#ifndef _USB_SERIAL_H_
+#define _USB_SERIAL_H_
+
+#if !defined (USBCON)
+#warning "Using legacy CDC core (non pluggable)"
+#else
+#include "Stream.h"
+#include "usbd_core.h"
+
+/**
+ * @brief Virtual serial terminal.
+ */
+class USBSerial : public Stream {
+public:
+    USBSerial(void);
+
+    void reenumerate();
+
+    void begin(void);
+    void begin(int);
+    void end(void);
+
+    virtual int available(void);
+    virtual int peek(void);
+    virtual void flush(void);
+    virtual int read(void);
+
+    int availableForWrite(void);
+    virtual size_t write(uint8_t);
```

Comment:
You overwrite only per-byte write method and use timer resource... But you can overwrite 
virtual size_t write(const uint8_t *buffer, size_t size);
method too, and just put received by method buffers into USB... without any timer usage!

Reply:
https://github.com/stm32duino/Arduino_Core_STM32/pull/344#issuecomment-427192572